src/components/content-tab-snapshots.tsx

```diff
@@ -70,6 +73,14 @@ function SnapshotRow( {
 		maxValue: 95,
 	} );
 
+	useEffect( () => {
+		const intervalId = setInterval( () => {
+			fetchSnapshotUsage(); // Function to fetch the latest data from the endpoint
+		}, 10000 );
+
+		return () => clearInterval( intervalId ); // Cleanup the interval on component unmount
+	}, [ fetchSnapshotUsage ] );
```

Comment:
I would avoid using an active listening. It not "wrong" and it works as expected, but I see it unnecessary and a trend that could  lead in adding more listeners all the time.

Maybe one request when the component mounts is enough?

```suggestion
	useEffect( () => {
		fetchSnapshotUsage();
	}, [ fetchSnapshotUsage ] );
```

My reasoning is that the backend will fail anyway and after it fails we can update the status of this button and disable it with the correct tooltip.

Reply:
Yeah that was my initial thinking; I have made these changes in https://github.com/Automattic/studio/pull/354/commits/fbd83095550af6a441a261c742cb799f9841bad5